test/test.js

```diff
@@ -872,45 +872,102 @@ describe('ParseMock', function(){
  */
 
   context('when object has beforeSave hook registered', function() {
-    function beforeSave() {
-      if (this.get("error")) {
+
+    function beforeSavePromise(request) {
+      var brand = request.object;
+      if (brand.get("error")) {
         return Parse.Promise.error("whoah");
       }
-      this.set('cool', true);
-      return Parse.Promise.as(this);
+      brand.set('cool', true);
+      return Parse.Promise.as(brand);
     }
 
-    it('runs the hook before saving the model and persists the object', function() {
-      ParseMockDB.registerHook('Brand', 'beforeSave', beforeSave);
+    it('runs the hook before saving the model and persists the object', function (done) {
+      ParseMockDB.registerHook('Brand', 'beforeSave', beforeSavePromise);
 
       var brand = new Brand();
       assert(!brand.has('cool'));
+      brand.set('nestedObject', { foo: 3 });
 
-      brand.save().then(savedBrand => {
-        assert(savedBrand.has('cool'));
+      brand.save().then(function (savedBrand) {
+        assert(savedBrand.has('cool'), 'saved brand doesn\'t have cool');
         assert(savedBrand.get('cool'));
-
-        var q = new Parse.Query(Brand);
-        q.first().then(queriedBrand => {
+        assert.equal(savedBrand.get('nestedObject').foo, 3);
+        new Parse.Query(Brand).first().then(function (queriedBrand) {
           assert(queriedBrand.has('cool'));
           assert(queriedBrand.get('cool'));
+          assert.equal(savedBrand.get('nestedObject').foo, 3);
+          done();
         });
       });
-    })
+    });
 
     it('rejects the save if there is a problem', function(done) {
-      ParseMockDB.registerHook('Brand', 'beforeSave', beforeSave);
+      ParseMockDB.registerHook('Brand', 'beforeSave', beforeSavePromise);
 
       var brand = new Brand({error: true});
 
-      brand.save().then(savedBrand => {
+      brand.save().then(function (savedBrand) {
         throw new Error("should not have saved")
-      }, error => {
+      }, function(error) {
         assert.equal(error, "whoah");
         done();
       });
-    })
-  })
+    });
+  });
+
+  context('when object has beforeDelete hook registered', function() {
+
+    var beforeDeleteWasRun;
+
+    beforeEach(function () {
```

Comment:
one last thing, would you mind putting all the parenthesis next to the function keywords as above (and also fix it below)